### PR TITLE
Notebook on Euler's algorithm

### PR DESCRIPTION
**Pull request by dtekinoglu**
@rossbar would you recommend me to make the reasoning behind the Euler's method more detailed? Besides, what might be the reason of tests being failed? @MridulS 

PR for [Issue #71](https://github.com/networkx/nx-guides/issues/71)

### CODE REVIEW

**Comment by dtekinoglu** [comment]
@rossbar @MridulS @dschult Could you please review this PR? I did not include `eulerize` method in the notebook to keep it simple. If you prefer otherwise, I can definetely add it. Any feedback will be greatly appreciated 😄

**Review by rossbar — COMMENTED** [review]
This is looking great, thanks @dtekinoglu!

As a first pass, I mostly focused on big-picture comments. I also made a few formatting changes in an effort to make the notebook better integrate with the website-building infrastructure. I'll cover both below:

### Content

Generally I think the content is great. I like using the classic 7-bridges of Konigsberg problem to introduce the concept of Eulerian paths and otherwise frame the discussion. At this stage, the thing I'd most like to see is to use NetworkX a little further up; specifically in the section where the 7-bridges problem is being introduced!

### Other changes

I made a few other changes to help get things going. Make sure to `git pull` before pushing up any of your own updates so that you have the locally (LMK if this causes problems!)

 - Google colab adds a lot of metadata cruft to notebooks, so I removed that in bb6c37d and e0dcf94
 - In order to make the notebook executable with the build system we use, we had to convert to `myst` format. I'll hold off on going into detail about it here; the most important thing is to answer *how do I get my notebooks into myst format*? The easiest way IMO is to use `jupytext`. For example, to convert from the `.md` format here to the `myst` format, I did `jupytext --from markdown --to myst content/algorithms/euler/euler.md`. The result of this command can be seen in the diff for c97cd66.
 - After the above conversion, there were some cells where you wanted to illustrate code but *not* execute it. As discussed in the meeting, the easiest way to do this is to change the cell type from `code-cell` to a non-executable directive, e.g. see d57c6a0
 - Lastly, I removed the bold formatting from all the headings just so that the styling matches the other tutorials :)

File: content/algorithms/euler/euler.md [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+<!-- #region id="9FhWLnEC0YYD" -->
+![image:part1](images/part1.png)
+<!-- #endregion -->
+
+<!-- #region id="RdAXIb8b0eHn" -->
+Euler observed that the choice of route inside each land mass is irrelevant. The only thing that matters is the sequence of bridges to be crossed in order. This observation allows us to abstract the problem even more. In the below graph, blue vertices represent the land masses and edges represent the bridges that connect them.
+<!-- #endregion -->
+
+<!-- #region id="2-C7-GbM0f8T" -->
+![image:part2](images/part2.png)
```
[the remark this answers is not in the document]
Comment (dtekinoglu) [reply content/algorithms/euler/euler.md:43]: Thanks for the suggestions @rossbar 

I am trying to visualize Königsberg graph, however, it is very hard to realize that the graph I constructed using `nx.draw()` actually represents Königsberg. I wrote the following snippet for it. At every run, nodes and edges are badly dispersed. Is there a better way for visualization? (e.g. to make it look more organized like in the image)

```
import networkx as nx
import matplotlib.pyplot as plt

G = nx.MultiGraph([("A","B"),("B","A"),("A","C"),("C","A"),("A","D"),("C","D"),("B","D")])

pos = nx.random_layout(G)
nx.draw_networkx_nodes(G, pos, label=True, node_color = 'b', node_size = 500, alpha = 1)
nx.draw_networkx_labels(G,pos, font_color="w")
ax = plt.gca()
for e in G.edges:
    ax.annotate("",
                xy=pos[e[0]], xycoords='data',
                xytext=pos[e[1]], textcoords='data',
                arrowprops=dict(arrowstyle="-", color="0.5",
                                shrinkA=5, shrinkB=5,
                                patchA=None, patchB=None,
                                connectionstyle="arc3,rad=rrr".replace('rrr',str(0.3*e[2])
                                ),
                                ),
                )
plt.axis('off')
plt.show()```

[the remark this answers is not in the document]
Comment (dtekinoglu) [reply content/algorithms/euler/euler.md:43]: An example output:

image

[the remark this answers is not in the document]
Comment (dschult) [reply content/algorithms/euler/euler.md:43]: My suggestion would be to make the layout manually.  A at (-1, 0), B at (0, -1) etc.  so the nodes are at the positions in the other diagram.  Using integer position values is simple. If you need more you can shift to including decimals, but most of my manual position dicts just use integers.

**Review by MridulS — APPROVED** [review]
This is great! Thanks for this @dtekinoglu :)

I have made a couple of changes to make this pass the linters by running nbqa-black and nbqa-pyupgrade on the notebook locally and updated the .md file. Also fixed some typos.

This looks ready to go to me :)

**Pull request merged**